From 702be3546c48eb19deb8dacb1d6eef9de9159538 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:28:30 -0400
Subject: [acpi] acpica: Add helper macros to extract bus/segment numbers from HEST table

Message-id: <20140719042829.19438.10783.stgit@gir.stowe>
Patchwork-id: 86017
O-Subject: [RHEL7.1 PATCH 55/81] ACPICA: Add helper macros to extract bus/segment numbers from HEST table.
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 4059a31063e9bf931222af23b8a3609773bf303f

commit 4059a31063e9bf931222af23b8a3609773bf303f
Author: Betty Dall <betty.dall@hp.com>
Date:   Mon Jan 13 12:17:47 2014 -0700

    ACPICA: Add helper macros to extract bus/segment numbers from HEST table.

    This change adds two macros to extract the encoded bus and segment
    numbers from the HEST Bus field.

    Signed-off-by: Betty Dall <betty.dall@hp.com>
    Signed-off-by: Bob Moore <robert.moore@intel.com>
    Signed-off-by: Lv Zheng <lv.zheng@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/acpi/actbl1.h b/include/acpi/actbl1.h
index 0bd750e..d46a51d 100644
--- a/include/acpi/actbl1.h
+++ b/include/acpi/actbl1.h
@@ -457,7 +457,7 @@ struct acpi_hest_aer_common {
  u8 enabled;
  u32 records_to_preallocate;
  u32 max_sections_per_record;
- u32 bus;
+ u32 bus;  /* Bus and Segment numbers */
  u16 device;
  u16 function;
  u16 device_control;
@@ -473,6 +473,14 @@ struct acpi_hest_aer_common {
 #define ACPI_HEST_FIRMWARE_FIRST        (1)
 #define ACPI_HEST_GLOBAL                (1<<1)
 
+/*
+ * Macros to access the bus/segment numbers in Bus field above:
+ *  Bus number is encoded in bits 7:0
+ *  Segment number is encoded in bits 23:8
+ */
+#define ACPI_HEST_BUS(bus)              ((bus) & 0xFF)
+#define ACPI_HEST_SEGMENT(bus)          (((bus) >> 8) & 0xFFFF)
+
 /* Hardware Error Notification */
 
 struct acpi_hest_notify {
-- 
1.7.1